### PR TITLE
Manage controller-gen via go.mod

### LINKED ISSUE #1719: Manage controller-gen via go.mod
**What would you like to be added**:

Keep the version in go.mod instead of the Makefile

**Why is this needed**:

To get upgrade PRs from the bot

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Manage controller-gen via go.mod and update it to 0.14.0.

#### Which issue(s) this PR fixes:

Fixes #1719 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/internal/tools/pinversion.go [hunk 1]
```diff
@@ -16,7 +16,11 @@ limitations under the License.
 
 package tools
 
-// Keep a reference  to code-generator so it's not removed by go mod tidy
+// Keep a reference to the code generators so they are not removed by go mod tidy
 import (
 	_ "k8s.io/code-generator"
+	// since verify will error o referencing a cmd package
```
Comment (tenzen-y) [inline hack/internal/tools/pinversion.go:22]:
```suggestion
	// since verify will error occur referencing a cmd package
```

Is this a typo?

**Review by tenzen-y — COMMENTED** [review]
/approve

just a nit.

File: hack/internal/tools/pinversion.go (hunk 1 above)
Comment (trasc) [reply hack/internal/tools/pinversion.go]: done , (it should have been "on" )

**Pull request merged**